# Sleep health and lifestyle

In this project we took the Sleep Health and Lifestyle Dataset in where you can find data on different people such as their profession, physical activity and sleep quality. We will take this dataset to know what are the things that affect sleep such as the amount of physical activity, the profession and the pressure to which each person is subjected.

## Import libraries

In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

## Load data

In [2]:
data = pd.read_csv(
    "/kaggle/input/sleep-health-and-lifestyle-dataset/Sleep_health_and_lifestyle_dataset.csv",
    index_col = "Person ID"
)
data.head()

Gender  Age            Occupation  Sleep Duration  Quality of Sleep  \
Person ID                                                                       
1           Male   27     Software Engineer             6.1                 6   
2           Male   28                Doctor             6.2                 6   
3           Male   28                Doctor             6.2                 6   
4           Male   28  Sales Representative             5.9                 4   
5           Male   28  Sales Representative             5.9                 4   

           Physical Activity Level  Stress Level BMI Category Blood Pressure  \
Person ID                                                                      
1                               42             6   Overweight         126/83   
2                               60             8       Normal         125/80   
3                               60             8       Normal         125/80   
4                               30             8        Obese         140/90   
5                               30             8        Obese         140/90   

           Heart Rate  Daily Steps Sleep Disorder  
Person ID                                          
1                  77         4200            NaN  
2                  75        10000            NaN  
3                  75        10000            NaN  
4                  85         3000    Sleep Apnea  
5                  85         3000    Sleep Apnea

## Questions for analysis

- Who has worse sleep quality, men or women?
- Is there a relationship between people's sleep quality and their profession?
- Does physical activity affect sleep?
- Which profession has the worst sleep quality?
- Which profession have people with the highest level of stress and the highest body mass index?
- In what age range are the most sleep disorders found?
- Does the number of steps per day affect sleep quality and body mass index?

Summary of the data set (using `.info()` method)

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 374 entries, 1 to 374
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Gender                   374 non-null    object 
 1   Age                      374 non-null    int64  
 2   Occupation               374 non-null    object 
 3   Sleep Duration           374 non-null    float64
 4   Quality of Sleep         374 non-null    int64  
 5   Physical Activity Level  374 non-null    int64  
 6   Stress Level             374 non-null    int64  
 7   BMI Category             374 non-null    object 
 8   Blood Pressure           374 non-null    object 
 9   Heart Rate               374 non-null    int64  
 10  Daily Steps              374 non-null    int64  
 11  Sleep Disorder           155 non-null    object 
dtypes: float64(1), int64(6), object(5)
memory usage: 38.0+ KB


Generate descriptive statistics (using `.describe()` method).

Descriptive statistics include those that summarize the central tendency, dispersion and shape of a dataset's distribution, excluding NaN values.

In [4]:
data.describe()

Age  Sleep Duration  Quality of Sleep  Physical Activity Level  \
count  374.000000      374.000000        374.000000               374.000000   
mean    42.184492        7.132086          7.312834                59.171123   
std      8.673133        0.795657          1.196956                20.830804   
min     27.000000        5.800000          4.000000                30.000000   
25%     35.250000        6.400000          6.000000                45.000000   
50%     43.000000        7.200000          7.000000                60.000000   
75%     50.000000        7.800000          8.000000                75.000000   
max     59.000000        8.500000          9.000000                90.000000   

       Stress Level  Heart Rate   Daily Steps  
count    374.000000  374.000000    374.000000  
mean       5.385027   70.165775   6816.844920  
std        1.774526    4.135676   1617.915679  
min        3.000000   65.000000   3000.000000  
25%        4.000000   68.000000   5600.000000  
50%        5.000000   70.000000   7000.000000  
75%        7.000000   72.000000   8000.000000  
max        8.000000   86.000000  10000.000000

## Ages analisys

Count the number of men and women

In [5]:
data["Gender"].value_counts()

# or

# data.Gender.value_counts()

Gender
Male      189
Female    185
Name: count, dtype: int64

> This gives us a general overview of the group, and we can say that the group is in proportion to the number of men and women. Proportion, with respect to the number of men and women.

## Classification by age

In the set we can find the ages of the persons, with respect to the ages we can find the following:

In [6]:
data["Age"].min()

27

In [7]:
data["Age"].max()

59

The lowest age is 27 years old
The highest age is 59 years old

What we can say is that the data set takes into account people who are already of working age, starting work or with only a few years in it, or people close to retiring. Let's find the average age in the data set.

In [8]:
data["Age"].mean()

42.18449197860963

The average age of the data set is 42 years and the standard deviation is 8.67, which means that in this set we can find very dispersed data, which may be due to the few data collected.

To support us in this analysis, a new column was added to the data set called Age_group in which the age of the people was classified as follows:

- **Age less than or equal to 40 years**: *Adult*.
- **Age over 40 years**: *OlderAdult*

In [9]:
data.loc[data["Age"] <= 40, 'AgeGroup'] = "Adult"
data.loc[data["Age"] >= 41, 'AgeGroup'] = "OlderAdult"

# View the first rows of the data set
data.head()

Gender  Age            Occupation  Sleep Duration  Quality of Sleep  \
Person ID                                                                       
1           Male   27     Software Engineer             6.1                 6   
2           Male   28                Doctor             6.2                 6   
3           Male   28                Doctor             6.2                 6   
4           Male   28  Sales Representative             5.9                 4   
5           Male   28  Sales Representative             5.9                 4   

           Physical Activity Level  Stress Level BMI Category Blood Pressure  \
Person ID                                                                      
1                               42             6   Overweight         126/83   
2                               60             8       Normal         125/80   
3                               60             8       Normal         125/80   
4                               30             8        Obese         140/90   
5                               30             8        Obese         140/90   

           Heart Rate  Daily Steps Sleep Disorder AgeGroup  
Person ID                                                   
1                  77         4200            NaN    Adult  
2                  75        10000            NaN    Adult  
3                  75        10000            NaN    Adult  
4                  85         3000    Sleep Apnea    Adult  
5                  85         3000    Sleep Apnea    Adult

In [10]:
# View the last rows of the data set
data.tail()

Gender  Age Occupation  Sleep Duration  Quality of Sleep  \
Person ID                                                             
370        Female   59      Nurse             8.1                 9   
371        Female   59      Nurse             8.0                 9   
372        Female   59      Nurse             8.1                 9   
373        Female   59      Nurse             8.1                 9   
374        Female   59      Nurse             8.1                 9   

           Physical Activity Level  Stress Level BMI Category Blood Pressure  \
Person ID                                                                      
370                             75             3   Overweight         140/95   
371                             75             3   Overweight         140/95   
372                             75             3   Overweight         140/95   
373                             75             3   Overweight         140/95   
374                             75             3   Overweight         140/95   

           Heart Rate  Daily Steps Sleep Disorder    AgeGroup  
Person ID                                                      
370                68         7000    Sleep Apnea  OlderAdult  
371                68         7000    Sleep Apnea  OlderAdult  
372                68         7000    Sleep Apnea  OlderAdult  
373                68         7000    Sleep Apnea  OlderAdult  
374                68         7000    Sleep Apnea  OlderAdult

Taking into account this ordering of the data, we have the following:

In [11]:
data['AgeGroup'].value_counts()

AgeGroup
OlderAdult    209
Adult         165
Name: count, dtype: int64

Finally, the average age for each gender is as follows:

In [12]:
data.groupby("Gender")["Age"].agg(["mean", "std"])

mean       std
Gender                     
Female  47.405405  8.093407
Male    37.074074  5.662006

We can see that there is a difference of 10 years in the average age of both genders, and we also note a large difference in the standard deviation, which is 3 years.

## Occupations analisys

In [13]:
data.groupby("Occupation")["Gender"].value_counts()

Occupation            Gender
Accountant            Female    36
                      Male       1
Doctor                Male      69
                      Female     2
Engineer              Female    32
                      Male      31
Lawyer                Male      45
                      Female     2
Manager               Female     1
Nurse                 Female    73
Sales Representative  Male       2
Salesperson           Male      32
Scientist             Female     4
Software Engineer     Male       4
Teacher               Female    35
                      Male       5
Name: count, dtype: int64

**We note the following:**

The occupations where there is more presence of women are:

- Accountant
- Engineering
- Manager
- Nursing
- Science
- Teacher

Occupations where men have more presence:

- Doctor
- Sales representative
- Sales
- Lawyer
- Software Engineer

It is important to mention that only in Engineering the difference is of only one person.

In [14]:
print(data["Sleep Duration"].min(), data["Sleep Duration"].max())

5.8 8.5


The average sleep duration between men and women is as follows:

In [15]:
data.groupby("Gender")[["Sleep Duration", "Quality of Sleep"]].mean()

Sleep Duration  Quality of Sleep
Gender                                  
Female        7.229730          7.664865
Male          7.036508          6.968254

We can see that in general, women have a better sleep than men not only in quantity but also in quality, to go deeper, we will obtain the same data but now in each of the professions of the set.

In [16]:
occupation_quality = data.groupby("Occupation")[["Sleep Duration", "Quality of Sleep"]].mean()
occupation_quality.sort_values(by = "Quality of Sleep")

Sleep Duration  Quality of Sleep
Occupation                                            
Sales Representative        5.900000          4.000000
Scientist                   6.000000          5.000000
Salesperson                 6.403125          6.000000
Software Engineer           6.750000          6.500000
Doctor                      6.970423          6.647887
Teacher                     6.690000          6.975000
Manager                     6.900000          7.000000
Nurse                       7.063014          7.369863
Accountant                  7.113514          7.891892
Lawyer                      7.410638          7.893617
Engineer                    7.987302          8.412698

In these tables we can see that the worst occupations with the worst quality and quantity of sleep are of sleep are:

- Sales Representative
- Scientist
- Sales

And those with the best quality and time of sleep are:

- Engineer
- Lawyer
- Accountant

In the professions with the worst quality and quantity of sleep there are more men than women.

## Health

Finally, we will look at how healthy each of the people in the different professions are, with this data we will be able to draw our conclusions and answer the questions we asked ourselves at the beginning of our analysis.

To know the body mass index by occupations.

In [17]:
occupation_bmi = data.groupby("Occupation")["BMI Category"].value_counts()
occupation_bmi.sort_values()

Occupation            BMI Category 
Lawyer                Normal Weight     1
Software Engineer     Overweight        1
                      Obese             1
Manager               Overweight        1
Teacher               Obese             1
Doctor                Normal Weight     2
Lawyer                Obese             2
                      Overweight        2
Software Engineer     Normal Weight     2
Sales Representative  Obese             2
Engineer              Overweight        3
                      Normal Weight     4
Doctor                Obese             4
Scientist             Overweight        4
Accountant            Normal Weight     5
                      Overweight        6
Teacher               Normal            6
Nurse                 Normal Weight     7
Accountant            Normal           26
Salesperson           Overweight       32
Teacher               Overweight       33
Lawyer                Normal           42
Engineer              Normal           5

Another important point to consider is to know if the number of steps per day affects health and sleep quality, to know this we will have the average daily steps, the level of physical activity, sleep duration and sleep quality in each of the professions.

In [18]:
occupation_sorted_by_qos = data.groupby("Occupation")[
    ["Daily Steps", "Physical Activity Level", "Sleep Duration", "Quality of Sleep"]
].mean()

occupation_sorted_by_qos.sort_values(by = "Quality of Sleep")

Daily Steps  Physical Activity Level  Sleep Duration  \
Occupation                                                                   
Sales Representative  3000.000000                30.000000        5.900000   
Scientist             5350.000000                41.000000        6.000000   
Salesperson           6000.000000                45.000000        6.403125   
Software Engineer     5800.000000                48.000000        6.750000   
Doctor                6808.450704                55.352113        6.970423   
Teacher               5957.500000                45.625000        6.690000   
Manager               5500.000000                55.000000        6.900000   
Nurse                 8057.534247                78.589041        7.063014   
Accountant            6881.081081                58.108108        7.113514   
Lawyer                7661.702128                70.425532        7.410638   
Engineer              5980.952381                51.857143        7.987302   

                      Quality of Sleep  
Occupation                              
Sales Representative          4.000000  
Scientist                     5.000000  
Salesperson                   6.000000  
Software Engineer             6.500000  
Doctor                        6.647887  
Teacher                       6.975000  
Manager                       7.000000  
Nurse                         7.369863  
Accountant                    7.891892  
Lawyer                        7.893617  
Engineer                      8.412698

## Conclusions

This data set serves to give us a general overview of each of the professions and how they affect people's health. I say general because one problem with this dataset is the amount of data it has, since it only has 307 rows, which are too few for the entire age range taken. This can be noticed when we obtain the average of the ages and the standard deviation, so that, at the end of our analysis, we can say that we need more data or, if necessary, reduce the age range of the respondents.

Moving on to each person's dream, we have the following:

Women in general terms have a better quality and quantity of sleep than men, either because women may have better stress management or because the women surveyed appear to be in professions where they talk to fewer people than men's professions.

We can follow this up by taking only the first five professions that have the best quality and quantity of sleep:

- Engineering
- Lawyer
- Accountant
- Nursing
- Manager

Of these professions, in three there are more women (Accountant, Nursing and Manager), in one there are more men (Lawyer) and in Engineering there is a difference of one person between the two.

Many of these professions are also repeated when we review the number of steps per day, we can see that the five professions where the most steps per day are taken are:

- Nursing
- Lawyer
- Accountant
- Doctor
- Salesperson

In most of these people we can see that walking is a fundamental part of them so it is not surprising that they have a high average number of steps per day and this is also seen in the health of sleep, because if we see the table of steps per day we can see that there is a relationship that the more steps are given the better sleep will be.

Although this same relationship is not noticeable when we talk about the body mass index, because from the above list we can see that in the following are many overweight people:

- Nursing
- Salesperson

Finally, the questions asked about the age of the respondents were not answered, since it was found that the range of ages is very high and the amount of data very few, so that the answers that can be found in this area can not be relevant to reality.